File: Travel Companion/Model/Firestore/FirestoreConstants.swift
//
//  FirestoreConstants.swift
//  Travel Companion
//
//  Created by Stefan Jaindl on 14.08.18.
//  Copyright © 2018 Stefan Jaindl. All rights reserved.
//

import Foundation

class FirestoreConstants {
    struct Collections {
        static let USERS = "users"
        static let PLACES = "places"
        static let PLANS = "plans"
        
        static let FLIGTHS = "fligths"
        static let PUBLIC_TRANSPORT = "publicTransport"
        static let HOTELS = "hotels"
        static let RESTAURANTS = "restaurants"
        static let ATTRACTIONS = "attractions"
    }
    
    struct Ids {
        struct Place {
            static let PLACE_ID = "placeId"
            static let LATITUDE = "latitude"
            static let LONGITUDE = "longitude"
            static let NAME = "name"
        }
        
        struct Plan {
            static let NAME = "name"
            static let PIN_NAME = "pinName"
            static let START_DATE = "startDate"
            static let END_DATE = "endDate"
            static let IMAGE_REFERENCE = "image"
        }
        
        struct User {
            static let UID = "uid"
            static let EMAIL = "email"
            static let NAME = "displayName"
            static let PROVIDER = "providerID"
            static let PHOTO_URL = "photoURL"
            static let PHONE = "phoneNumber"
        }
    }    
}

File: shared/src/commonMain/kotlin/com.sjaindl.travelcompanion/api/firestore/FireStoreConstants.kt
package com.sjaindl.travelcompanion.api.firestore

class FireStoreConstants {
    object Collections {
        const val users = "users"
        const val places = "places"
        const val plans = "plans"

        const val hotels = "hotels"
        const val restaurants = "restaurants"
        const val attractions = "attractions"

        const val photos = "photos"
    }

    object Ids {
        object Place {
            const val placeId = "place_id"
            const val id = "id"
            const val name = "name"
            const val icon = "icon"
            const val rating = "rating"
            const val reference = "reference"
            const val scope = "scope"
            const val userRatingsTotal = "user_ratings_total"
            const val vicinity = "vicinity"

            const val types = "types"

            const val geometry = "geometry"
            const val location = "location"
            const val latitude = "latitude"
            const val longitude = "longitude"

            const val photos = "photos"
            const val height = "height"
            const val width = "width"
            const val photoReference = "photo_reference"
            const val htmlAttributions = "html_attributions"

            const val plusCode = "plus_code"
            const val compound_code = "compound_code"
            const val global_code = "global_code"
        }

        object Plan {
            const val name = "name"
            const val pinName = "pinName"
            const val startDate = "startDate"
            const val endDate = "endDate"
            const val imageReference = "image"

            const val path = "path"
        }

        object Plannable {
            const val notes = "notes"
        }

        object User {
            const val userId = "uid"
            const val email = "email"
            const val displayName = "displayName"
            const val providerId = "providerID"
            const val photoUrl = "photoURL"
            const val phoneNumber = "phoneNumber"
        }
    }
}
